### ISSUE TITLE
[Improvement] An empty state placeholder for the forms 

### ISSUE THREAD

**Issue by aniri** [code4romania/mon-vot-android-kotlin#100]
See the new UI here: https://www.figma.com/file/lww9lcabUpamTZg8zCExI8/MV-2.0-Prototype?node-id=292%3A93 . 

An empty state placeholder for the forms in case the internet connection is poor and the forms can't be retrieved.

English text: `We could not download the forms. Please check your internet connection and tap the button below to try again.`
Romanian text: `Formularele nu au putut fi descărcate. Te rugăm să verifici conexiunea la internet și să apeși pe butonul de mai jos pentru a încerca din nou.` 

Also show an error in case there is no internet connection when pressing the retry button (same error as when trying to login when there is no internet connection).

<img width="375" alt="Empty state forms" src="https://user-images.githubusercontent.com/1358055/91643325-cad23d00-ea3a-11ea-9804-eaaf2297baa4.png">

**Comment by aniri** [comment]
On iOS a retry button was added to fix this :)

**Comment by LarisaHabliuc** [comment]
I have created the UI for this [here](https://www.figma.com/file/lww9lcabUpamTZg8zCExI8/MV-2.0-Prototype?node-id=292%3A93).

**Comment by lukstbit** [comment]
@aniri Should this empty state be shown every time the API call for forms fails? At the moment when the API call for forms fails we fallback to the database stored forms. If the database is empty(like when the user first starts the app) then we should definitely show the empty state UI. But what should happen if we do have valid forms already stored in the database?

**Comment by aniri** [comment]
@lukstbit for now, the empty state should be shown when there are no forms (nothing retrieved from the api / api error and no forms in db). We'll have to think about the other case with api error, but old forms in db in the future :smiley: thanks for the idea!

**Comment by lukstbit** [comment]
@aniri I'm looking into implementing this tomorrow.

While fixing #246 I also saw that there's no notification to the user(in the county/polling station selection screen) when the app starts for the first time and there's no connection(the exact situation like here). The user is left with an empty county spinner and a button telling him to choose a county. There should be an issue for this as well.

**Comment by aniri** [comment]
@lukstbit yep, you are right. I think we can fix that also with a similar empty state (just with another text). Would you have time to include that also in this issue or should we make a separate one?

**Comment by lukstbit** [comment]
@aniri  Make another issue. The UI is almost the same but the implementation will be different.

**Comment by lukstbit** [comment]
@aniri I don't think I will be able to make a PR on this before the elections. I keep getting into issues to make everything around the forms list work correctly. That screen is very important  in the app and I don't want to implement something rushed to introduce bugs there and break functionality.

**Comment by aniri** [comment]
@lukstbit no problem! yep, you are right, it's best not to rush things. we're also planning on doing a big rewrite of the error handling part, started here https://github.com/code4romania/mon-vot-android-kotlin/pull/221 . So we'll figure everything out then, the empty states also :D

Thanks a lot for all the issues you crushed and made ready for the elections!! :rocket:
